Riot/Modules/LocationSharing/UserLocationService.swift

```diff
@@ -111,34 +115,33 @@ class UserLocationService: UserLocationServiceProtocol {
         })
     }
     
+    private func getExistingDeviceBeaconSummaries() -> [MXBeaconInfoSummaryProtocol] {
+        guard let userId = self.session.myUserId else {
+            return []
+        }
+        
+        return self.session.locationService.getBeaconInfoSummaries(for: userId).filter { summary in
+            return self.isDeviceBeaconInfoSummary(summary) && summary.isActive
```

Comment:
Does this need to be weak? Also if I am reading this correctly we are calling `isDeviceBeaconInfoSummary` twice when restoring, once when `getExistingDeviceBeaconSummaries` and second time when `didReceiveDeviceNewBeaconInfoSummary` is called for the same session. Should both methods be responsible for making this check?

Reply:
> Does this need to be weak?

You mean: `private let session: MXSession` needs to be weak?

> if I am reading this correctly we are calling isDeviceBeaconInfoSummary twice when restoring

In fact in `didReceiveBeaconInfoSummary` where we are listenning to every beacon info summaries we are checking `isDeviceBeaconInfoSummary` and then call `didReceiveDeviceNewBeaconInfoSummary` when we have a new BeaconInfoSummary that belongs to the device . But there is no further call to `isDeviceBeaconInfoSummary` inside `didReceiveDeviceNewBeaconInfoSummary`.

Reply:
> You mean: `private let session: MXSession` needs to be weak?

Ah its a `filter`, not an escaping callback, so ignore this.

> In fact in `didReceiveBeaconInfoSummary` where we are listenning to every beacon info summaries we are checking `isDeviceBeaconInfoSummary` and then call `didReceiveDeviceNewBeaconInfoSummary` when we have a new BeaconInfoSummary that belongs to the device . But there is no further call to `isDeviceBeaconInfoSummary` inside `didReceiveDeviceNewBeaconInfoSummary`.

You're right, I was confused by the similarity of `didReceiveBeaconInfoSummary` and `didReceiveDeviceNewBeaconInfoSummary` (or rather not noticing the second method)

Reply:
No problem I admit the that the two function names are closed